contracts/upgradeable_contracts/erc20_to_native/HomeBridgeErcToNative.sol

```diff
@@ -0,0 +1,82 @@
+pragma solidity 0.4.24;
+import "../../libraries/SafeMath.sol";
+import "../../libraries/Message.sol";
+import "../BasicBridge.sol";
+import "../../upgradeability/EternalStorage.sol";
+import "../../IBlockReward.sol";
+import "../../ERC677Receiver.sol";
+import "../BasicHomeBridge.sol";
+import "../ERC677Bridge.sol";
+
+
+contract HomeBridgeErcToNative is EternalStorage, BasicBridge, BasicHomeBridge {
+
+    function () public payable {
+        require(msg.value > 0);
+        require(msg.data.length == 0);
+        require(withinLimit(msg.value));
+        setTotalSpentPerDay(getCurrentDay(), totalSpentPerDay(getCurrentDay()).add(msg.value));
+        setTotalBurntCoins(totalBurntCoins().add(msg.value));
+        address(0).transfer(msg.value);
+        emit UserRequestForSignature(msg.sender, msg.value);
+    }
+
+    function initialize (
+        address _validatorContract,
+        uint256 _dailyLimit,
+        uint256 _maxPerTx,
+        uint256 _minPerTx,
+        uint256 _homeGasPrice,
+        uint256 _requiredBlockConfirmations,
+        address _blockReward
+
+    ) public returns(bool)
+    {
+        require(!isInitialized());
+        require(_validatorContract != address(0));
+        require(_homeGasPrice > 0);
+        require(_requiredBlockConfirmations > 0);
+        require(_minPerTx > 0 && _maxPerTx > _minPerTx && _dailyLimit > _maxPerTx);
+        addressStorage[keccak256(abi.encodePacked("validatorContract"))] = _validatorContract;
+        uintStorage[keccak256(abi.encodePacked("deployedAtBlock"))] = block.number;
+        uintStorage[keccak256(abi.encodePacked("dailyLimit"))] = _dailyLimit;
+        uintStorage[keccak256(abi.encodePacked("maxPerTx"))] = _maxPerTx;
+        uintStorage[keccak256(abi.encodePacked("minPerTx"))] = _minPerTx;
+        uintStorage[keccak256(abi.encodePacked("gasPrice"))] = _homeGasPrice;
+        uintStorage[keccak256(abi.encodePacked("requiredBlockConfirmations"))] = _requiredBlockConfirmations;
+        setBlockRewardContract(_blockReward);
+        setInitialize(true);
+
+        return isInitialized();
+    }
+
+    function getBridgeMode() public pure returns(bytes4 _data) {
+        return bytes4(keccak256(abi.encodePacked("erc-to-native-core")));
+    }
+
+    function blockRewardContract() public view returns(IBlockReward) {
+        return IBlockReward(addressStorage[keccak256(abi.encodePacked("blockRewardContract"))]);
+    }
+
+    function totalBurntCoins() public view returns(uint256) {
+        return uintStorage[keccak256(abi.encodePacked("totalBurntCoins"))];
+    }
+
+    function setBlockRewardContract(address _blockReward) internal {
+        require(_blockReward != address(0));
```

Comment:
We need also to be sure that the address is a contract account.

Reply:
Thanks, fixed!